third_party/indexed-db-as-promised/src/sync-promise.js

```diff
@@ -0,0 +1,194 @@
+export default class SyncPromise {
+  constructor(resolver) {
+    if (!isFunction(resolver)) {
+      throw new TypeError('Must pass resolver function');
+    }
+
+    this._state = PendingPromise;
+    this._value = [];
+
+    doResolve(
+      this,
+      adopter(this, FulfilledPromise),
+      adopter(this, RejectedPromise),
+      { then: resolver }
+    );
+  }
+
+  then(onFulfilled, onRejected) {
+    onFulfilled = isFunction(onFulfilled) ? onFulfilled : void 0;
+    onRejected = isFunction(onRejected) ? onRejected : void 0;
+
+    return this._state(
+      this._value,
+      onFulfilled,
+      onRejected
+    );
+  }
+
+  catch(onRejected) {
+    return this.then(void 0, onRejected);
+  }
+
+  static resolve(value) {
+    const Constructor = this;
```

Comment:
awesome work on the library. for my own knowledge why do this instead of referencing the class name? just being defensive in case the reference is assigned somewhere else and to not make it nominal?